third_party/proton/tutorials/intra_kernel/README.md

```diff
@@ -0,0 +1,117 @@
+
+**Workflow**:
+1. **Generate TTGIR dump files**:
+   ```bash
+   ../../scripts/dump_ttgir.sh python3 add_override.py
+   ```
+   Creates original TTGIR files in `ttgir_dump/` directory
+
+2. **Insert profiling instrumentation**:
+   ```bash
+   ./insert_proton_records
+   ```
+   Modifies TTGIR files by adding `proton.record` operators at profiling points
+
+3. **Execute with TTGIR override**:
+   ```bash
+   TRITON_ALWAYS_COMPILE=1 TRITON_KERNEL_OVERRIDE=1 TRITON_OVERRIDE_DIR=ttgir_dump python3 add_override.py
+   ```
+   - `TRITON_ALWAYS_COMPILE=1`: Forces recompilation on each run
+   - `TRITON_KERNEL_OVERRIDE=1`: Enables TTGIR override mechanism
+   - `TRITON_OVERRIDE_DIR=ttgir_dump`: Specifies directory with modified TTGIR files
+
+### 2. Proton DSL Approach (`matmul_dsl.py`)
+
+**Use Case**: Native profiling integration for Gluon DSL kernels
+
+**Example**: Matrix multiplication using NVIDIA Hopper architecture features (WGMMA, TMA)
+
+
+**Command Line Options**:
+```bash
+# Timeline trace mode (default)
+python3 matmul_dsl.py
+
+# Operation measurement mode
+python3 matmul_dsl.py --op-measure
+
+# Enable warp sampling with specific warp IDs
+python3 matmul_dsl.py --warp-sampling --warp-ids "0,1,2,3"
+
+# High accuracy profiling (recommended)
+python3 matmul_dsl.py --increase-accuracy
+```
+
+## Understanding Timeline Traces
+
+### Time Representation
+- **Scope Duration**: Displayed in cycles for precise measurement
+- **Threadblock Start Times**: Measured in nanoseconds using global timing
+- **Chrome Trace Format**: Assumes 1GHz GPU frequency for consistent time units (ns)
+
+### Circular Buffer System
+- **Backend Storage**: Uses circular buffer for runtime profiling on each CTA
+- **Buffer Overflow**: When full, earlier events are dropped with warnings in trace generation
+- **Event Window**: Displays sliding window of most recent events in timeline
+
+### Finalize Time Measurement
+- **Definition**: Captures `Finalize Time` when kernel execution completes
+- **Meaning**: Shows overhead of dumping profiling data from buffer to global memory (appears as a field in Chrome trace viewer tab)
+
+## Configuration Options
+
+### Profiling Accuracy
+
+| Option | Description | Use Case |
+|--------|-------------|----------|
+| `clock32` | Records events in 32-bit clock format for lower overhead | normal kernels (<4 seconds @ 1GHz) |
+| `time_shift` | Deducts constant profiling overhead from timeline trace | Mitigate Proton runtime overhead for cleaner traces |
+| `sched_stores` | Provides more cycle-accurate operation latency measurement | Accurate single operation latency measure |
+| `sched_barriers` | Constrains AMD instruction scheduling within proton scopes | AMD GPU profiling |
+
+### Buffer Configuration
+
+| Parameter | Options | Description |
+|-----------|---------|-------------|
+| `buffer_type` | `shared_mem`, `global_mem` | Storage location for profiling buffer (only `shared_mem` currently supported) |
```

Comment:
Let's just take out `global_mem` for now because other options we don't support are not mentioned here